docs/elasticsearch.rst

```diff
@@ -63,9 +65,16 @@ expected to return a ``doc_dict``, so just add the ``username`` to that.
 Building the new index
 ''''''''''''''''''''''
 Once you've made the change, you'll need to build a new index which uses
-that new mapping, so you'll have to update the hash at the top of the file.
-This can just be a random alphanumeric string.  This will trigger a preindex
-as outlined in the `Indexes` section.
+that new mapping. Updating index name in the mapping file triggers HQ to
+create the new index with new mapping and reindex all data, so you'll
+have to update the index hash and alias at the top of the mapping file.
+The hash suffix to the index can just be a random alphanumeric string and
+is usually the date of the edit by convention. The alias should also be updated
+likewise or simply suffixed with ``_new``, for example ``xforms_new``, so that
+production operations continue to use the old alias pointing to existing index.
+This will trigger a preindex as outlined in the `Indexes` section. In subsequent commits
+alias can be flipped back to what it was, for example ``xforms``. Changing the alias
+name doesn't trigger a reindex.
```

Comment:
when the alias is renamed in code will that automatically get updated in ES? Will the old (`xforms_new`) alias still exist or does it get removed?

Reply:
If the old alias remains then I would suggest a different suffix other than `_new` like the date or something

Reply:
Good idea, I have updated the docs.

Reply:
I don't see the changes